infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/K8sContainerResolver.java

```diff
@@ -37,11 +36,16 @@
  */
 public class K8sContainerResolver {
 
+
+  private final int maxContainerNameLength = 63; //K8S container name limit
+  private final String pluginName;
   private final CheContainer cheContainer;
   private final List<ChePluginEndpoint> containerEndpoints;
 
-  public K8sContainerResolver(CheContainer container, List<ChePluginEndpoint> containerEndpoints) {
+  public K8sContainerResolver(
+      CheContainer container, String pluginName, List<ChePluginEndpoint> containerEndpoints) {
```

Comment:
It doesn't really change a lot but I would not separate container and containerEndpoints with plugin name parameter. I think it would be a bit better to have `K8sContainerResolver(String pluginName, CheContainer container, List<ChePluginEndpoint> containerEndpoints)`